sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/DiagnosticScopeLiveTests.cs

```diff
@@ -16,14 +16,27 @@ namespace Azure.Messaging.ServiceBus.Tests.Diagnostics
     [NonParallelizable]
     public class DiagnosticScopeLiveTests : ServiceBusLiveTestBase
     {
+        private TestDiagnosticListener _listener;
+
+        [SetUp]
+        public void Setup()
+        {
+            _listener = new TestDiagnosticListener(EntityScopeFactory.DiagnosticNamespace);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _listener.Dispose();
```

Comment:
These tests occasionally have errors that indicate they are stomping on each others events, but I'm not sure how that is possible given that these are being run in NonParallelizable mode. Trying explicit teardown to see if it helps.

Reply:
yay, ambient diagnostic state!